src/core/server/saved_objects/service/lib/repository.security_extension.test.ts

```diff
@@ -0,0 +1,1896 @@
+      const objA = {
+        ...obj1,
+        type: MULTI_NAMESPACE_CUSTOM_INDEX_TYPE, // replace the type to a mult-namespace type for this test to be thorough
+        namespaces: multiNamespaceObjNamespaces, // include multiple spaces
+      };
+      const objB = { ...obj2, namespaces: ['ns-3'] }; // use a different namespace than the options namespace;
+      const optionsNamespace = 'ns-4';
+
+      await bulkGetSuccess(client, repository, registry, [objA, objB], {
+        namespace: optionsNamespace,
+      });
+
+      expect(mockSecurityExt.checkAuthorization).toHaveBeenCalledTimes(1);
+      const expectedActions = ['bulk_get'];
+      const expectedSpaces = new Set([optionsNamespace, ...objA.namespaces, ...objB.namespaces]);
+      const expectedTypes = new Set([objA.type, objB.type]);
+
+      const {
+        actions: actualActions,
+        spaces: actualSpaces,
+        types: actualTypes,
+      } = mockSecurityExt.checkAuthorization.mock.calls[0][0];
+
+      expect(actualActions).toEqual(expectedActions);
+      expect(setsAreEqual(actualSpaces, expectedSpaces)).toBeTruthy();
+      expect(setsAreEqual(actualTypes, expectedTypes)).toBeTruthy();
+    });
+
+    test(`calls enforceAuthorization with action, type map, and auth map`, async () => {
+      const objA = {
+        ...obj1,
+        type: MULTI_NAMESPACE_CUSTOM_INDEX_TYPE, // replace the type to a mult-namespace type for this test to be thorough
+        namespaces: multiNamespaceObjNamespaces, // include multiple spaces
+      };
+      const objB = { ...obj2, namespaces: ['ns-3'] }; // use a different namespace than the options namespace;
+      const optionsNamespace = 'ns-4';
+
+      setupCheckAuthorized();
+
+      await bulkGetSuccess(
+        client,
+        repository,
+        registry,
+        [objA, objB],
+        { namespace: optionsNamespace } // foo-namespace is alredy in multiNamespaceObjNamespaces
+        // (though we could leave it in as a duplicate - those are NOT removed, ToDo: remove duplicate namespace checks?)
+      );
+
+      expect(mockSecurityExt.checkAuthorization).toHaveBeenCalledTimes(1);
+      expect(mockSecurityExt.enforceAuthorization).toHaveBeenCalledTimes(1);
+      expect(mockSecurityExt.enforceAuthorization).toHaveBeenCalledWith(
+        expect.objectContaining({
+          action: 'bulk_get',
+        })
+      );
+
+      const expectedTypesAndSpaces = new Map([
+        [objA.type, new Set([optionsNamespace, ...objA.namespaces])],
+        [objB.type, new Set([optionsNamespace, ...objB.namespaces])],
+      ]);
+
+      const { typesAndSpaces: actualTypesAndSpaces, typeMap: actualTypeMap } =
+        mockSecurityExt.enforceAuthorization.mock.calls[0][0];
+
+      expect(mapsAreEqual(actualTypesAndSpaces, expectedTypesAndSpaces)).toBeTruthy();
+      expect(actualTypeMap).toBe(authMap); // ToDo? reference comparison ok, object is frozen
```

Comment:
Is it ok to simplify this check as an object reference equality? Alternatively, I could write a function to compare this type of Map.